nifi-framework-bundle/nifi-framework-extensions/nifi-py4j-framework-bundle/nifi-python-extension-api/src/main/python/src/nifiapi/properties.py

```diff
@@ -293,6 +298,106 @@ def __add_resource_definition(self, gateway, resource_definition, builder):
         builder.identifiesExternalResource(cardinality, types[0], types[1:])
 
 
+def _collect_descriptor_metadata(java_context):
+    descriptor_list = []
+    descriptor_lookup = {}
+    value_lookup = {}
+
+    try:
+        java_properties = java_context.getProperties()
+    except AttributeError:
```

Comment:
What can trigger an `AttributeError`? Should this be logged or otherwise thrown?